File: src/dart/hprose/lib/src/class_manager.dart
/**********************************************************\
|                                                          |
|                          hprose                          |
|                                                          |
| Official WebSite: http://www.hprose.com/                 |
|                   http://www.hprose.net/                 |
|                   http://www.hprose.org/                 |
|                                                          |
\**********************************************************/
/**********************************************************\
 *                                                        *
 * class_manager.dart                                     *
 *                                                        *
 * hprose class manager for Dart.                         *
 *                                                        *
 * LastModified: Feb 8, 2014                              *
 * Author: Ma Bingyao <andot@hprose.com>                  *
 *                                                        *
\**********************************************************/
part of hprose;

abstract class HproseClassManager {
  static final Map<ClassMirror, String> aliasCache = new Map<ClassMirror, String>();
  static final Map<String, ClassMirror> classCache = new Map<String, ClassMirror>();

  static void register(ClassMirror cm, String alias) {
    aliasCache[cm] = alias;
    classCache[alias] = cm;
  }

  static String getClassAlias(ClassMirror cm) {
    return aliasCache[cm];
  }

  static ClassMirror getClass(String alias) {
    return classCache[alias];
  }
}

File: src/java/java5/src/hprose/io/ClassManager.java
/**********************************************************\
|                                                          |
|                          hprose                          |
|                                                          |
| Official WebSite: http://www.hprose.com/                 |
|                   http://www.hprose.net/                 |
|                   http://www.hprose.org/                 |
|                                                          |
\**********************************************************/
/**********************************************************\
 *                                                        *
 * ClassManager.java                                      *
 *                                                        *
 * ClassManager for Java.                                 *
 *                                                        *
 * LastModified: May 6, 2011                              *
 * Author: Ma Bingyao <andot@hprose.com>                  *
 *                                                        *
\**********************************************************/
package hprose.io;

import java.util.concurrent.ConcurrentHashMap;

public final class ClassManager {
    private static final ConcurrentHashMap<Class<?>, String> classCache1 = new ConcurrentHashMap<Class<?>, String>();
    private static final ConcurrentHashMap<String, Class<?>> classCache2 = new ConcurrentHashMap<String, Class<?>>();

    private ClassManager() {
    }

    public static void register(Class<?> type, String alias) {
        classCache1.put(type, alias);
        classCache2.put(alias, type);
    }

    public static String getClassAlias(Class<?> type) {
        return classCache1.get(type);
    }

    public static Class<?> getClass(String alias) {
        return classCache2.get(alias);
    }

    public static boolean containsClass(String alias) {
        return classCache2.containsKey(alias);
    }
}